javaslang/src/main/java/javaslang/collection/RedBlackTree.java

```diff
@@ -68,7 +68,7 @@
 
     static <T extends Comparable<? super T>> RedBlackTree<T> ofAll(Iterable<? extends T> values) {
         Objects.requireNonNull(values, "values is null");
-        return ofAll((Comparator<? super T> & Serializable) T::compareTo, values);
+        return ofAll(Comparators.naturalComparator(), values);
```

Comment:
Something wrong with cast to `(Comparator<> & Serializable)`, need to be investigated